### PR TITLE
Move VS Code integration tests to /test directory

### PR DESCRIPTION
**Pull request by charisk**
Moves VS Code integration tests out of `/src` and into `/test`. 

## Checklist
N/A:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/stories/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "lib": ["ES2021", "dom"],
     "jsx": "react",
     "sourceMap": true,
-    "rootDir": "..",
+    "rootDir": "../../..",
```
Comment (norascheuch) [inline extensions/ql-vscode/src/stories/tsconfig.json:10]: I don't know about stories, but doesn't have this side effects?

File: extensions/ql-vscode/test/tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-    "../src/vscode-tests/factories/db-config-factories.ts"
-  ],
+  "include": ["**/*.ts", "../src/**/*.ts"],
   "exclude": [],
   "compilerOptions": {
     "noEmit": true,
     "resolveJsonModule": true,
-    "rootDir": ".."
+    "rootDir": "..",
+    "jsx": "react"
```
Comment (norascheuch) [inline extensions/ql-vscode/test/tsconfig.json:9]: Why did you add this here?

**Review by norascheuch — APPROVED** [review]
✨ 😮  Amazing changes!

File: extensions/ql-vscode/src/stories/tsconfig.json (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/stories/tsconfig.json:10]: This basically says "look at the whole of `extensions/ql-vscode` when compiling. This is needed because we use stuff from `/test` inside `/src/stories`. Long term we might want to move `/src/stories` into `/test/stories`.

Comment (norascheuch) [reply extensions/ql-vscode/src/stories/tsconfig.json:10]: 👍

File: extensions/ql-vscode/test/tsconfig.json (hunk 2 above)
Comment (charisk) [reply extensions/ql-vscode/test/tsconfig.json:9]: Because we now need to compile jsx files as part of `/test` (because we compile all of `/extensions/ql-vscode` which includes `/src`)
